lsl/imaging/utils.py

```diff
@@ -1544,19 +1544,23 @@ def bm_image(self, center=(0,0), term=None, weighting='natural', local_fraction=
             return [self._gen_img(b, center=center, weighting=weighting, local_fraction=local_fraction, robust=robust, taper=taper) for b in self.bm]
 
 
-def build_gridded_image(data_set, size=80, res=0.50, wres=0.10, pol='XX', chan=None, im=None, verbose=True):
+def build_gridded_image(data_set, size=80, res=0.50, wres=0.10, pol='XX', uv_size=-1.0, chan=None, im=None, verbose=True):
     """
     Given a :class:`lsl.imaging.data.VisibilityDataSet` object, build an aipy.img.ImgW 
     object of gridded uv data which can be used for imaging.  The ImgW object 
     itself is returned by this function to make it more versatile.
     """
-    
+    # Catch UV resolution setting for specific calculation of projection to override default 'size'
+    # Assumes user has properly calculated uvres such that round(uvres/res) results in a usable final image size.
+    if uv_size != 1.0:
+        size = uv_size
+        
```

Comment:
Wait, if `uv_size` is not the default we make `size` equal to `uv_size`?  Am I missing something here?

Reply:
I will change this to None instead of -1 (typo in code). I had left it in this weird format because I was not sure if build_gridded_image() was unique to imageIDI.py or not. I figured it was not, so I wanted to preserve the 'size' variable in case it was hanging around inside some other function. The variable 'uv_size' is used in my imageIDI as a calculated quantity given the input parameters of image size and pixel resolution upstream. So, this 'catch' would act to delineate my imageIDI calls to this function and swap out the default size variable with my calculated uv_resolution. It is pretty hammy, but we can change it if you want.

Reply:
`build_gridded_image` isn't specific to imageIDI.py but I still don't get `size` vs. `uv_size` since they appear to be the same thing.  Maybe this is more of a distinction in imageIDI.py where `size` is image size and `uv_size` is the grid size?